pyproject.toml

```diff
@@ -21,28 +21,30 @@ google-search-results = "*"
 pytest = "*"
 llama-index = "~0.9.0"
 duckduckgo-search = "*"
-crewai = {extras = ["tools"], version = ">=0.22.5"}
 # metagpt = "*" # Commented out because requires super old version of langchain, and conflicts with crewai.
 replicate = "*"
 typer = "^0.9.0"
 web3 = "^6.14.0"
 mypy = "^1.8.0"
-types-requests = "^2.31.0.20240106"
 black = "^23.12.1"
 google-cloud-functions = "^1.16.0"
 google-cloud-resource-manager = "^1.12.0"
 poetry = "^1.7.1"
 poetry-plugin-export = "^1.6.0"
 functions-framework = "^3.5.0"
 cron-validator = "^1.0.8"
-prediction-market-agent-tooling = "^0.13.0"
+prediction-market-agent-tooling = {git = "https://github.com/gnosis/prediction-market-agent-tooling.git", rev = "evan/mech-client-deps-compat"}
 pydantic-settings = "^2.1.0"
 autoflake = "^2.2.1"
 isort = "^5.13.2"
 markdownify = "^0.11.6"
 tavily-python = "^0.3.1"
 # TODO remove when PR is merged
 microchain-python = {git = "https://github.com/evangriffiths/microchain.git", rev = "evan/token-tracker"}
+types-requests = "2.31.0.0"
+crewai = {version = "0.11.0", extras = ["tools"]}
```

Comment:
crewai conflict with mech-client deps

Reply:
I assume this will also be reverted back to v0.28 once PMAT adds mech-client as dependency?

Reply:
mech-client is super very strictly versioned. if they do this in other repos as well then it's even worse 🙈  I think 
it's worth a try to replace these `==` (exact version) with `^` (any newer version except for the versions with breaking changes). 

Maybe there still will be some conflicts but a lot less, at least hopefully.

https://github.com/valory-xyz/mech-client/blob/main/pyproject.toml#L27

![Screenshot by Dropbox Capture](https://github.com/gnosis/prediction-market-agent/assets/15619339/50fe2c7e-0974-42ff-9da0-677128251b31)

Reply:
Unfortunately doesn't help 😞 

The conflicts come from further down the dependency tree
- here: https://github.com/valory-xyz/open-aea/blob/1bb984ba44b028d941266ffeb13558bb4f32645d/plugins/aea-ledger-ethereum/setup.py#L46
- and here https://github.com/valory-xyz/open-aea/blob/1bb984ba44b028d941266ffeb13558bb4f32645d/setup.py#L63

So there's strictness to fix first in https://github.com/valory-xyz/open-aea and in the mech-client repo